README.md

```diff
@@ -85,12 +117,415 @@ return (
 );
 ```
 
-## Install
+</details>
 
-```shell
-npm i @gravity-ui/onboarding
+## Onboarding configuration
+
+You can configure onboarding
+
+```typescript jsx
+const onboardingOptions = {
+  config: {
+    presets: {/**/},
+    baseState: {/**/}, // initial state for current user
+    getProgressState: () => {/**/}, // function to load user progress
+    onSave: {
+      // functions to save user state
+      state: (state) => {/**/},
+      progress: (progress) => {/**/},
+    },
+    showHint: (state) => {
+      /**/
+    }, // optional. function to shoe hint. Only for vanilla js usage
+    logger: {
+      // optional. you can specify custom logger
+      level: 'error' as const,
+      logger: {
+        debug: () => {/**/},
+        error: () => {/**/},
+      },
+    },
+    debugMode: true, // optional. true will show a lot of debug messages. Recommended for dev environment
+    plugins: [/**/], // optional. you can use existing plugins or write your own
+    hooks: {
+      // optional. you can subscribe to onboarding events
+      showHint: ({preset, step}) => {/**/},
+      stepPass: ({preset, step}) => {/**/},
+      addPreset: ({preset}) => {/**/},
+      beforeRunPreset: ({preset}) => {/**/},
+      runPreset: ({preset}) => {/**/},
+      finishPreset: ({preset}) => {/**/},
+      beforeSuggestPreset: ({preset}) => {/**/},
+      beforeShowHint: ({stepData}) => {/**/},
+      stateChange: ({state}) => {/**/},
+      hintDataChanged: ({state}) => {/**/},
+      closeHint: ({hint}) => {/**/},
+      init: () => {/**/},
+      wizardStateChanged: ({wizardState}) => {/**/},
+    },
+  },
+};
+```
+
+---
+
+For default preset you can specify properties:
+
+```typescript jsx
+const onboardingOptions = {
+  config: {
+    presets: {
+      createProject: {
```

Comment:
`createPreset`?

Reply:
createPreset is only for better typing. Added comment about that